src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -48,22 +48,22 @@ public void destroy() {
 
 	@Given("a user clicks on Edit Registration Information link from Patient dashboard")
 	public void loadRegistrationSummaryPage() {
-		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage.goToRegistrationSummary().waitForPage();
+		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage.goToRegistrationSummary();
 	}
 
 	@Then("the system loads Registration Summary Page")
 	public void systemLoadsRegistrationSummaryPage() {
-		assertPage(registrationSummaryPage.waitForPage());
+		assertPage(registrationSummaryPage);
 	}
 
 	@When("a user clicks on Edit link from Registration Summary Page")
 	public void loadEditDemographicsSection() {
-		registrationEditSectionPage = (RegistrationEditSectionPage) registrationSummaryPage.clickOnEditDemographics().waitForPage();
+		registrationEditSectionPage = (RegistrationEditSectionPage) registrationSummaryPage.clickOnEditDemographics();
 	}
 
 	@Then("the system loads edit demographics section")
 	public void systemLoadsEditDemographicsSection() {
-		assertPage(registrationEditSectionPage.waitForPage());
+		assertPage(registrationEditSectionPage);
```

Comment:
we need to be careful when dealing with assertPage(), it tends to apply the assertion logic on the page in the context so fast and usually fails the run when the page is not waited for to load. This usually happens on Firefox browser so I would retain the waitForPage() function on pages under assertion.